### PR TITLE
docs: add a deployment doc for Forcerelay/Axon

### PR DESCRIPTION
**Pull request by ashuralyk**

### CODE REVIEW

**Review by blckngm — APPROVED** [review]

File: docs/deployment.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+deploy from the repo (Yarn needed):
+```bash
+$ git clone https://github.com/synapseweb3/ibc-solidity-contract
+$ cd ibc-solidity-contract
+$ yarn install
+$ yarn compile
+$ AXON_HTTP_RPC_URL=https://rpc-alphanet-axon.ckbapp.dev yarn migrate
+```
+ 
+after the deployment, contracts `OwnableIBCHandler` and `ICS20TransferERC20` are in your output, let's record the addresses of them as **<IBC_HANDLER_ADDRESS>** and **<TRANSFER_ERC20_ADDRESS>** for a later use.
```
Comment (Xcodes-chain) [inline docs/deployment.md:25]: 1. The console.log in the 1_deploy_contracts.js script returns too much print content, making it inconvenient to filter `OwnableIBCHandler` and `ICS20TransferERC20`. It is recommended to simplify; or return the values ​​of the two agreed addresses required for configuration.
```
 return {
    ibcHandlerAddress: ibcHandler.address,
    ics20Transfer address: ics20Transfer.address
  };
```

File: docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+we have compiled all of the latest 4 ckb contracts, named `Connection`, `Channel`, `Packet` and `sudt-transfer`, in the repo of Forcerelay, please make sure there's enough CKB capacity in your ckb address **<your_ckb_address>** and follow the below steps (ckb-cli needed):
+```bash
+$ git clone https://github.com/synapseweb3/forcerelay
+$ cd forcerelay/tools/ibc-test/contracts/deployment
+$ ./upload.sh connection https://testnet.ckbapp.dev/ <your_ckb_address>
+$ ./upload.sh channel https://testnet.ckbapp.dev/ <your_ckb_address>
+$ ./upload.sh packet https://testnet.ckbapp.dev/ <your_ckb_address>
+$ ./upload.sh sudt https://testnet.ckbapp.dev/ <your_ckb_address>
+```
```
Comment (Xcodes-chain) [inline docs/deployment.md:37]: Does metadata also need to be deployed?

Comment (Xcodes-chain) [reply docs/deployment.md:37]: it is necessary to supplement and update the lock.args corresponding to the deployment contract account.

File: docs/deployment.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+websocket_addr = "wss://rpc-alphanet-axon.ckbapp.dev/ws"
+rpc_addr = "https://rpc-alphanet-axon.ckbapp.dev"
+contract_address = <IBC_HANDLER_ADDRESS>
+restore_block_count = 10000
+key_name = "relayer_axon_wallet"
+store_prefix = "forcerelay"
+```
+
+write your keys of Axon and CKB in two different files privately, named **<your_ckb_private_file>** and **<your_axon_private_file>**, and then, import them to the installed Forcerelay/Axon:
+```bash
```
Comment (Xcodes-chain) [inline docs/deployment.md:117]: Specific cases where the private key may need to be supplemented are for relayer_ckb_wallet and relayer_axon_wallet in the configuration.

**Review by 15168316096 — COMMENTED** [review]
Need to reply to the content of the comment and improve some points

File: docs/deployment.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+## Generate Port Id
+port id in Axon for transfer use is always named `transfer`, but in CKB is different, it's the lock hash of deployed `sudt-transfer` cell, which `args` can be temporarily set as ZERO:
+```toml
+# lock script of sudt-transfer
+code_hash = <SUDT_TRANSFER_TYPE_ID>
+hash_type = type
+args = 0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+```
+
+calculate blake2b hash of above lock script, and let's record it as **<ST_CELL_LOCK_HASH>**.
```
Comment (Xcodes-chain) [inline docs/deployment.md:139]: You may need some preliminary knowledge. It is recommended to use lumos to obtain the codehash of the script.
import {utils} from '@ckb-lumos/lumos'
const cell = await rpc.getLiveCell(txHash, index)
const codeHash = utils.computeScriptHash(cell.output.type)

File: docs/deployment.md (hunk 2 above)
Comment (ashuralyk) [reply docs/deployment.md:37]: > Does metadata also need to be deployed?

In real case, metadata cell is not deployed that way like these IBC contracts do

File: docs/deployment.md (hunk 3 above)
Comment (ashuralyk) [reply docs/deployment.md:117]: these can be specific and do not change, they just the name of the wallet, I can leave a comment here as a hint

**Pull request merged**